File: frontend/src/algorithmCode/searching/jumpSearch/jumpSearch.c
#include <stdio.h>
#include <math.h>

int jumpSearch(int arr[], int n, int target) {
    int step = sqrt(n);
    int prev = 0;

    while (arr[(step < n ? step : n) - 1] < target) {
        prev = step;
        step += sqrt(n);
        if (prev >= n)
            return -1;
    }

    for (int i = prev; i < (step < n ? step : n); i++) {
        if (arr[i] == target)
            return i;
    }

    return -1;
}

int main() {
    int arr[] = {1, 3, 5, 7, 9, 12, 15, 18};
    int n = sizeof(arr) / sizeof(arr[0]);
    int target = 12;

    int result = jumpSearch(arr, n, target);
    printf("Found at index: %d\n", result);
    return 0;
}

File: frontend/src/algorithmCode/searching/jumpSearch/jumpSearch.java
public class JumpSearch {
    public static int jumpSearch(int[] arr, int target) {
        int n = arr.length;
        int step = (int)Math.floor(Math.sqrt(n));
        int prev = 0;

        while (prev < n && arr[Math.min(step, n) - 1] < target) {
            prev = step;
            step += (int)Math.floor(Math.sqrt(n));
            if (prev >= n) return -1;
        }

        for (int i = prev; i < Math.min(step, n); i++) {
            if (arr[i] == target)
                return i;
        }

        return -1;
    }

    public static void main(String[] args) {
        int[] arr = {1, 3, 5, 7, 9, 12, 15, 18};
        int target = 12;

        int result = jumpSearch(arr, target);
        System.out.println("Found at index: " + result);
    }
}
